lib/scripts.js

```diff
@@ -148,28 +168,33 @@ module.exports = function (gulp, gutil) {
            * example: angular.module("my.module", ["some.dependencies"])
            */
 
-          moduleFiles[module] = [filePath].concat(moduleFiles[module] || []);
-          moduleDependencies[module] = is().empty(dependencies) ?
-              ['ng'] : dependencies;
+            if (moduleFiles[module] === undefined || moduleFiles[module].dev === undefined) {
+                moduleFiles[module] = { dev : [filePath], min : [filePath] };
+            } else {
+                moduleFiles[module].dev.push(filePath);
+                moduleFiles[module].min.push(filePath);
+            }
+            moduleDependencies[module] = is().empty(dependencies) ? ['ng'] : dependencies;
         } else {
           /*
            * found module look up
            * example: angular.module("my.module")
            */
-          if (moduleFiles[module] === undefined) {
-            moduleFiles[module] = [filePath];
-          } else {
-            if (moduleFiles[module].indexOf(filePath) === -1) {
-              moduleFiles[module].push(filePath);
+            if (moduleFiles[module] === undefined || moduleFiles[module].dev === undefined) {
+                moduleFiles[module] = { dev : [filePath], min : [filePath] };
+            } else {
+                if (moduleFiles[module].dev.indexOf(filePath) === -1) {
+                    moduleFiles[module].dev.push(filePath);
+                    moduleFiles[module].min.push(filePath);
+                }
             }
-          }
         }
       }
     }
 
     return gulp.src(['**/*.js', '!lib/**/*.js'], {cwd: src})
       .pipe(isWatch ? plumber()() : gutil.noop())
-      .pipe(uglify()({compress: false, mangle: false}))
+      .pipe(uglify()({compress: true, mangle: true}))
```

Comment:
Uglify here is used to get content in right format in downstream files. I have enabled compress and mangle to keep it consistent within the file. 

Change in variable names (mangle) doesn't cause errors downstream files.

There is a slight performance hit, which I think is valid for better readability.